mypyc/test-data/run-valuetype.test

```diff
@@ -0,0 +1,336 @@
+        self.x: Final = x
+
+    def __hash__(self) -> int:
+        return hash(self.x)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, A):
+            return NotImplemented
+        return self.x == other.x
+
+    def __len__(self) -> int:
+        return 1
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, A):
+            return NotImplemented
+        return self.x < other.x
+
+    def __gt__(self, other: object) -> bool:
+        if not isinstance(other, A):
+            return NotImplemented
+        return self.x > other.x
+
+    def __le__(self, other: object) -> bool:
+        if not isinstance(other, A):
+            return NotImplemented
+        return self.x <= other.x
+
+    def __ge__(self, other: object) -> bool:
+        if not isinstance(other, A):
+            return NotImplemented
+        return self.x >= other.x
+
+    def __contains__(self, item: object) -> bool:
+        return isinstance(item, i32) and item == self.x
+
+    def __bool__(self) -> bool:
+        return self.x != 0
+
+    def __str__(self) -> str:
+        return f"A({self.x})"
+
+    def __repr__(self) -> str:
+        return f"A({self.x})"
+
+def test_direct_dunders() -> None:
+    a = A(1, 2)
+    assert a.__hash__() == hash(1)
+    assert a.__eq__(A(1, 2))
+    assert not a.__eq__(A(2, 3))
+    assert a.__len__() == 1
+    assert a.__lt__(A(2, 3))
+    assert a.__le__(A(2, 3))
+    assert not a.__gt__(A(2, 3))
+    assert not a.__ge__(A(2, 3))
+    assert a.__contains__(1)
+    assert not a.__contains__(2)
+    assert a.__bool__()
+    assert not A(0, 1).__bool__()
+    assert a.__str__() == "A(1)"
+    assert a.__repr__() == "A(1)"
+
+def test_value_type_in_key() -> None:
+    a = A(1, 2)
+    d = {a: 1}
+    assert d[A(1,5)] == 1
+
+def test_value_type_in_val() -> None:
+    a = A(1, 2)
+    d = {1: a}
+    assert d[1].x == 1
+    assert d[1] == A(1, 2)
+
+def test_value_type_in_set() -> None:
+    a = A(1, 2)
+    s = {a}
+    assert A(1, 2) in s
+
+def test_hash() -> None:
+    a = A(1, 2)
+    assert hash(a) == hash(1)
+
+def test_eq() -> None:
+    assert A(1, 2) == A(1, 2)
+    assert A(1, 2) != A(2, 3)
+    assert A(1, 2) != 1
+    assert 1 != A(1, 2)
+
+def test_bool() -> None:
+    assert bool(A(1, 2))
+    assert not bool(A(0, 2))
+
+def test_len() -> None:
+    assert len(A(1, 2)) == 1
+
+def test_str() -> None:
+    assert str(A(1, 2)) == "A(1)"
+    assert repr(A(1, 2)) == "A(1)"
+
+def test_cmp() -> None:
+    a = A(1, 2)
+    b = A(2, 3)
+    assert a < b
+    assert a <= b
+    assert b > a
+    assert b >= a
+    assert not a > b
+    assert not a >= b
+    assert not b < a
+    assert not b <= a
+    assert 1 in a
+    assert 2 not in a
+
+[case testValueTypeLifetime]
+from typing import final, Final, NamedTuple
+from mypy_extensions import mypyc_attr
+
+class A(NamedTuple):
+    n: str
+
+@final
+@mypyc_attr(value_type=True)
+class B:
```

Comment:
Test that we generate an error if `@final` is omitted but a class is declared as a value type.

Reply:
Due the previous comment, I will relax this constraint, so value types can be non final classes. But they still need to be immutable.